src/problem.jl

```diff
@@ -185,39 +185,72 @@ function JumpProblem(prob, aggregator::AbstractAggregatorAlgorithm, jumps::JumpS
        !is_spatial(aggregator) # check if need to flatten
         prob, maj = flatten(maj, prob, spatial_system, hopping_constants; kwargs...)
     end
-    ## Constant Rate Handling
-    t, end_time, u = prob.tspan[1], prob.tspan[2], prob.u0
-    if (typeof(jumps.constant_jumps) <: Tuple{}) && (maj === nothing) &&
-       !is_spatial(aggregator) # check if there are no jumps
-        disc = nothing
-        constant_jump_callback = CallbackSet()
-    else
-        disc = aggregate(aggregator, u, prob.p, t, end_time, jumps.constant_jumps, maj,
-                         save_positions, rng; spatial_system = spatial_system,
-                         hopping_constants = hopping_constants, kwargs...)
-        constant_jump_callback = DiscreteCallback(disc)
-    end
 
-    iip = isinplace_jump(prob, jumps.regular_jump)
+    ## Constant and variable rate handling
+    t, end_time, u = prob.tspan[1], prob.tspan[2], prob.u0
 
-    ## Variable Rate Handling
-    if typeof(jumps.variable_jumps) <: Tuple{}
+    if length(jumps.variable_jumps) == 0 && (length(jumps.constant_jumps) == 0) &&
+       (maj === nothing) && !is_spatial(aggregator)
+        # check if there are no jumps
+        new_prob = prob
+        variable_jump_callback = CallbackSet()
+        cont_agg = JumpSet().variable_jumps
+        disc_agg = nothing
+        constant_jump_callback = CallbackSet()
+    elseif typeof(aggregator) <: Coevolve
```

Comment:
Can we add a trait `supports_variablerates` which defaults to `false` but `Coevolve` overloads to `true`? Since we'd ultimately like to support other vrj supporting algorithms.

Reply:
Yes, that's a good idea.